js/utils/currency.js

```diff
@@ -115,3 +124,92 @@ export function formatCurrency(amount, currency, locale = app.settings.get('lang
 
   return formattedCurrency;
 }
+
+let exchangeRates = {};
+
+export function fetchExchangeRates(options = {}) {
```

Comment:
@jjeffryes You mentioned that you intended on fetching the exchange rates at order time to ensure the most accurate price. A couple of thoughts on that:

1.) Currently, we're fetching the exchange rates every 15 seconds (which is also how often the server fetches them), so a fetch-on-demand is at best going to get you rates that are 14.9999 seconds fresher than the cached ones available via getExchangeRate(). And, since our interval of fetching will not be in sync with the servers, it is likely the cached ones are even fresher than that and a re-fetch may return you the exact same values as the cached one.

So, I'm not sure how critical it is to fetch-on-demand.

2.) If you do decide to carry on and fetch-on-demand, I would recommend you do so via fetchExchangeRates() (and then convertCurrency() and/or convertAndFormatCurrency()). That will fetch all the exchange rates, however, that's probably more efficient than calling the api for individual rates, since if you need to convert between two fiat values, you would need to make two API calls which, with all things considered, will be heavier than calling the single bulkier one.